Formula/o/otpclient.rb

```diff
@@ -0,0 +1,35 @@
+class Otpclient < Formula
+  desc "C/GTK3 OTP client that supports both TOTP and HOTP"
+  homepage "https://github.com/paolostivanin/OTPClient/wiki"
+  url "https://github.com/paolostivanin/OTPClient/archive/refs/tags/v3.2.1.tar.gz"
+  sha256 "f926155ef104e4965561e668a2e5128cdb7ed19af49ba7e693b580883fdae048"
+  license "GPL-3.0-only"
+
+  depends_on "cmake" => :build
+  depends_on "pkg-config" => :build
+  depends_on "glib"
+  depends_on "gtk+3"
+  depends_on "jansson"
+  depends_on "libcotp"
+  depends_on "libgcrypt"
+  depends_on "libpng"
+  depends_on "libsecret"
+  depends_on "protobuf-c"
+  depends_on "qrencode"
+  depends_on "zbar"
+
+  def install
+    inreplace "CMakeLists.txt",
+      "pkg_check_modules(UUID REQUIRED uuid>=2.34.0)",
+      "pkg_check_modules(UUID REQUIRED uuid>=1.0)"
+    mkdir "build" do
+      system "cmake", "-S", "..", "-B", ".", *std_cmake_args
+      system "make"
+      system "make", "install"
+    end
+  end
+
+  test do
+    system "true"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
The CLI executable `otpclient-cli` requires `~/.config/otpclient.cfg` to function. This path is [hard-coded](https://github.com/paolostivanin/OTPClient/blob/ec8dcc15ca29d3668aee6d4b90def07ac013e1f3/src/cli/main.c#L204) and not possible to change with CLI arguments during test. I cannot figure out a way to test it without modifying `~/.config/otpclient.cfg` so far.

Reply:
`$HOME` is set to the test directory, so it should be enough to make a `.config` directory and add the file.

Reply:
i made a small test case with `expect`